### ISSUE TITLE
Jos on käännöksia ja jaetaan master-dokumentissa lohkoja osiin, niin rikkoo käännöksen

### ISSUE THREAD

**Issue by vesal** [TIM-JYU/TIM#3349]
Jos on tehnyt ensin masteriiin ison lohkon, sitten 
kääntää masterin ja sitten lisää masteriin #- merkkejä, niin 
käännökseen syntyy useita lohkoja ja jää myös se iso lohko.

Tuohon ei liene mitään helppoa ratkaisua.

**Comment by saviit** [comment]
Nopealla vilkaisulla näyttäisi siltä, että manuaalisesti lohkottamalla (#- merkeillä) tuo alkuperäinen lohko-id säilyy ns. master-dokussa, ja sen lisäksi tulee uusia lohko-id:tä. Sitten tuolla synchronize_translations -funktiossa (timApp/document/translation/synchronize_translations.py: 8) lisätään käännösdokusta puuttuvat lohkot lohko-id:n perusteella, mutta master-dokussa muuttuneita lohkoja ei muuteta käännöksessä vaan vain merkitään muuttuneiksi (Translation out of date).

Eikös tämä siis toimi ihan niin kuin on tarkoituskin? Mielestäni tästä on ollut aikaisemminkin puhetta, että noihin käännöksessä oleviin jo käännettyihin lohkoihin ei master-muutosten osalta saa mennä koskemaan muuten kuin lisäämällä tuon muutosmerkinnän (ettei menetetä käännöksessä jotain). Ja sitten on käännösversion ylläpitäjän harteilla korjata muutokset käännökseen?

**Comment by vesal** [comment]
Joo, toimii kuten on "määriteltykin", mutta aika työläs oli esittää käyttäjälle että alas
nyt sitten siellä käännösodkussa niitä valmiiksi käännettyjä poistamaan ja kopioimaan
uusin suomi-lohkojen päälle.  Mutta tosiaan haastavaa tehdä muulla tavalla kun pitäisi
jo käännettyyn saada niitä lohkomerkintöjä.

Toinen mikä tuli vastaan oli että olisi halunnut linkit menevän vastaaviin /en-US dokuihin, mutta
ei helppoa tämäkään koska voisi olla ettei koskaan edes tule sitä käännettyä versiota.

**Comment by saviit** [comment]
> Toinen mikä tuli vastaan oli että olisi halunnut linkit menevän vastaaviin /en-US dokuihin, mutta
> ei helppoa tämäkään koska voisi olla ettei koskaan edes tule sitä käännettyä versiota.

Tuon voisi kyllä lisätä sinne jatkokehityskorttiin (muistaakseni ei vielä ole), ja samaa ideaa noudattaen voisi selvitellä saisiko käännökseen tulevat viitelohkot osoittamaan oletuksena käännettyyn versioon (jos sellainen löytyy). Toki silloin pitäisi varmaan olla joku asetus millä tuon saisi kytkettyä (helposti) pois päältä ja sama noiden linkkien kanssa, jos käyttäjä ei haluakaan viitata muihin käännöksiin.

**Comment by vesal** [comment]
Entä voisiko tehdä niin, että jos on käännöksiä ja jakaa jo käännetyn lohkon useampaan osaan, niin siitä
tulisi kysymys että haluatko tosiaan näin ja sitten myös linkki ohjeeseen, missä neuvotaan miten jakaminen
sitten käsitellään käännöksen  puolella.  Tuo korjaus on kuitenkin kohtuullisen haastava ja koehenkilö
ei oikein meinannut siitä selvitä.  Ongelmia aiheutti erityissesti vaihtelu sen välillä että oliko #- vai ###-otsikkoa.

En muista tehtiinkö siitä atomien hävittämisestä ja siitä että olisi aina #- lohkon erottimena korttia?  Mutta
se pikkuisen suoraviivaistaisi tuota korjausoperaatiotakin.  Silloin voisi suoraan miettimättä copy/pasteta
jo käännettyjä lohkoja jakamisessa syntyneiden suomi-lohkojen päälle.

**Comment by saviit** [comment]
Atomilohkoista oli tosiaan jo kortti #3235
